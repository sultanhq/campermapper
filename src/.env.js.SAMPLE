export const GMAPS_API_KEY = 'YOUR_GOOGLEMAPS_API_KEY_HERE'
export const FIREBASE_OPTIONS = {
    apiKey: "{YOUR_FIREBASE_API_KEY_HERE}",
    authDomain: "{YOUR_DOMAIN_HERE}.firebaseapp.com",
    databaseURL: "{YOUR_DATABASE_URL_HERE}",
    }
