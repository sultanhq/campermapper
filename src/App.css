.App {
  position: absolute;
}

.App-logo {
  animation: App-logo-spin infinite 30s linear;
  height: 80%;
  padding: 5px;
}

.App-header {
  background-color: #222;
  height: 15vh;
  width: 100%;
  color: white;
  top: 0;
  position: fixed;
  display: flex;
  flex-direction: row;
}

.Menu {
  /*position: fixed;*/
  margin-left: left;
  /*top: 15vh;*/
  /*height: 5vh;*/
  width: 100vw;
  background-color: #777;
}

.h2 {
  height: 100%;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(360deg);
  }
  to {
    transform: rotate(0deg);
  }
}

.View {
  display: flex;
  flex-direction: row;
  margin-top: 15vh;
}

.siteListContainer {
  overflow: auto;
  left: 0;
  width: 30%;
}

.mapContainer {
  position: fixed;
  right: 0;
  width: 70%;
  /*margin-top: 10px*/
}

.site {
  list-style-type: none;
}

.Name {
  display: table;
  margin-left: left;
}

.Price {
  display: table;
  margin-left: left;
}

.Button {
  display: table;
  margin-left: left;
}
