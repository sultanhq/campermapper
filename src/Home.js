import React, { Component } from 'react';

export class Home extends Component {

  render(){
    return(
      <a
        className='Home'
        href={'/'}>
        Home
      </a>
    );
  }
};
