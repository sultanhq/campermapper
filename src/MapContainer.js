import React, {
  Component
} from 'react';
import {
  Gmaps,
  Marker
} from 'react-gmaps';
const params = {
  v: '3.exp',
  key: process.env.REACT_APP_GMAPS_API_KEY
};

export class MapContainer extends Component {
    constructor() {
    super();
    this.state = {
        siteData: {},
        coords: {
            latitude: 54.7115618,
            longitude: -6.0656706
        },
        zoom: {
            value: 6
        },
        selectedSite: null,
    };
}

  onMapCreated(map) {
    map.setOptions({
      disableDefaultUI: false,
      streetViewControl: false
    });
  }

  centerMap() {
    this.map.setCenter({
          lat: this.lat,
          lng: this.lng
        })
  }

  componentDidUpdate(nextProps) {
    const newSite = this.props.selectedSite !== nextProps.selectedSite
    if (newSite){
      this.setState({
        coords: {
          latitude: Object.values(this.props.sites.sites)[this.props.selectedSite].latitude,
          longitude: Object.values(this.props.sites.sites)[this.props.selectedSite].longitude
        },
        zoom: {
          value: 10
        },
      })
    }
  }

  render() {
    const data = this.props.sites.sites
    const createMarkers = Object.values(data).map((data, index) =>
      <Marker key={index}
        lat={data.latitude}
        lng={data.longitude}
        onClick={this.centerMap}
        />
    );

    return (
      <div className='map'>
        <Gmaps id='gmaps'
        width={'70vw'}
        height={'80vh'}
        lat={this.state.coords.latitude}
        lng={this.state.coords.longitude}
        zoom={this.state.zoom.value}
        loadingMessage={'Map Loading'}
        params={params}
        onMapCreated={this.onMapCreated}>
        { createMarkers }
        </Gmaps>
      </div>
    );
  };

};
