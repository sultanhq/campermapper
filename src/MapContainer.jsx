import React, {
  Component
} from 'react';
import {
  Gmaps,
  Marker
} from 'react-gmaps';
// import {
//   GMAPS_API_KEY
// } from './.env.js';


const params = {
  v: '3.exp',
  key: process.env['GMAPS_API_KEY']
};

export class MapContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      coords: {
        lat: 54.7115618,
        lng: -6.0656706
      },
      zoom: {value: 6},
    };
  }

  onMapCreated(map) {
    map.setOptions({
      disableDefaultUI: false,
      streetViewControl: false
    });
  }

  centerMap() {
    this.map.panTo({
      lat: this.lat,
      lng: this.lng
    })
  }

  render() {

    const data = this.props.sites

    const createMarkers = data.map((data) =>
      <Marker key={data.id}
        lat={data.lat}
        lng={data.lng}
        label={"" + data.id}
        onClick={this.centerMap}
        />
    );

    return (
      <div className='map'>
        <Gmaps id='gmaps'
        width={'70vw'}
        height={'80vh'}
        lat={this.state.coords.lat}
        lng={this.state.coords.lng}
        zoom={this.state.zoom.value}
        loadingMessage={'Map Loading'}
        params={params}
        onMapCreated={this.onMapCreated}>
        { createMarkers }
        </Gmaps>
      </div>
    );
  };

};
