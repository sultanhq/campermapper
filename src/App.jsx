import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import SiteData from './siteData.json'
import { SiteList } from '../src/siteList';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to Camper Mapper</h2>
        </div>
        <p className="App-intro">
        </p>
        <SiteList
          sites={SiteData.sites}
          />
      </div>
    );
  }
}

export default App;
