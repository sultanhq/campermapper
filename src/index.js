import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import './index.css';
// import { FIREBASE_OPTIONS } from './.env.js';
import * as firebase from 'firebase';

const FIREBASE_OPTIONS = {
apikey:process.env.REACT_APP_FIREBASE_API_KEY,
authDomain:process.env.REACT_APP_FIREBASE_AUTHDOMAIN,
databaseURL:process.env.REACT_APP_FIREBASE_DATABASEURL,
}
console.log(process.env)
firebase.initializeApp(FIREBASE_OPTIONS);

ReactDOM.render(
  <App />,
  document.getElementById('root')
);
