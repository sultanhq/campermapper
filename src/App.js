import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { SiteList } from '../src/SiteList';
import { MapContainer } from '../src/MapContainer';
import * as firebase from 'firebase';

class App extends Component {

  constructor(){
    super();
    this.state = {
      siteData: {
        "sites": [

        ]
      },

      selectedSite: null,
    };
    this.moveMap = this.moveMap.bind(this)
  }

  moveMap = (e) => {
    this.setState({selectedSite: e.id,
    zoom: {value: 10}})
  }

  componentDidMount() {
    const rootRef = firebase.database().ref().child('databases');
    const siteDataRef = rootRef.child('siteData');
      siteDataRef.on('value', snap => {
        this.setState({
          siteData: snap.val()
        });
    });
  }

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to Camper Mapper</h2>
        </div>
        <div className="View">
          <div className='siteListContainer'>
            <SiteList
              moveMap={this.moveMap}
              sites={this.state.siteData.sites}
            />
          </div>
          <div className='mapContainer'>
            <MapContainer
              sites={this.state.siteData.sites}
              selectedSite={this.state.selectedSite}
              />
          </div>
        </div>
      </div>
    );
  };

}


export default App;
