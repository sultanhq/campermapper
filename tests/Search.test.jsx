import React from 'react';
import renderer from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils'; // ES6
import {Search} from '../src/Search';

import { shallow } from 'enzyme'
import sinon from 'sinon'

describe('Search', () => {
	it ('a has a Search div', () => {
		const component = renderer.create(
      <div>
        <form
        className="searchForm"
        onSubmit={[Function]}>
        <input
          label="Search"
          onChange={[Function]}
          type="text"
          value="" />
        <input
          type="submit"
          value="search!" />
        </form>
      </div>
    ).toJSON();
		expect(component).toMatchSnapshot();
	});

  it ('search returns JSON', () => {
    const onMessageSend = jest.fn()

    const wrapper = shallow(
      React.createElement(Search, {
        onMessageSend
      })
    )

    wrapper.find('input').simulate('change', {target: {value: 'A test message'} })
    wrapper.find('form').simulate('submit', { preventDefault(){} })

    expect(onMessageSend).toHaveBeenCalledWith('A test message')

  });

});
