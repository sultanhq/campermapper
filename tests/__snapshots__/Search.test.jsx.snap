exports[`Search a has a Search div 1`] = `
<div>
  <form
    className="searchForm"
    onSubmit={
      Array [
        [Function],
      ]
    }>
    <input
      label="Search"
      onChange={
        Array [
          [Function],
        ]
      }
      type="text"
      value="" />
    <input
      type="submit"
      value="search!" />
  </form>
</div>
`;
